test/framework/src/main/java/org/elasticsearch/cluster/routing/RoutingNodesHelper.java

```diff
@@ -47,4 +49,7 @@ private RoutingNodesHelper() {}
         return shards;
     }
 
+    public static Stream<ShardRouting> asStream(IndexShardRoutingTable indexShardRoutingTable) {
```

Comment:
Maybe a comment on why this is in `:test:framework` rather than `:server`, to avoid someone undoing most of this change in a follow-up.